src/mappings/subgraphService.ts

```diff
@@ -20,3 +22,148 @@ export function handleRewardsDestinationSet(event: RewardsDestinationSet): void
+
+    // update indexer
+    let indexer = Indexer.load(indexerID)!
+    indexer.allocatedTokens = indexer.allocatedTokens.plus(event.params.tokens)
+    indexer.totalAllocationCount = indexer.totalAllocationCount.plus(BigInt.fromI32(1))
+    indexer.allocationCount = indexer.allocationCount + 1
+    indexer.save()
+
+    // update provision
+    let provision = createOrLoadProvision(event.params.indexer, event.address, event.block.timestamp)
+    provision.tokensAllocated = provision.tokensAllocated.plus(event.params.tokens)
+    provision.totalAllocationCount = provision.totalAllocationCount.plus(BigInt.fromI32(1))
+    provision.allocationCount = provision.allocationCount + 1
+    provision.save()
+
+    // update graph network
+    graphNetwork.totalTokensAllocated = graphNetwork.totalTokensAllocated.plus(event.params.tokens)
+    graphNetwork.allocationCount = graphNetwork.allocationCount + 1
+    graphNetwork.activeAllocationCount = graphNetwork.activeAllocationCount + 1
+    graphNetwork.save()
+
+    // update subgraph deployment
+    let deployment = createOrLoadSubgraphDeployment(subgraphDeploymentID, event.block.timestamp)
+    deployment.stakedTokens = deployment.stakedTokens.plus(event.params.tokens)
+    deployment.save()
+
+    // create allocation
+    let allocation = new Allocation(allocationID)
+    allocation.indexer = indexerID
+    allocation.provision = provision.id
+    allocation.creator = event.transaction.from
+    allocation.activeForIndexer = indexerID
+    allocation.activeForProvision = provision.id
+    allocation.subgraphDeployment = subgraphDeploymentID
+    allocation.allocatedTokens = event.params.tokens
+    allocation.effectiveAllocation = BigInt.fromI32(0)
+    allocation.createdAtEpoch = event.params.currentEpoch.toI32()
+    allocation.createdAtBlockHash = event.block.hash
+    allocation.createdAtBlockNumber = (
+        addresses.isL1 ? event.block.number : graphNetwork.currentL1BlockNumber!
+    ).toI32()
+    allocation.queryFeesCollected = BigInt.fromI32(0)
+    allocation.queryFeeRebates = BigInt.fromI32(0)
+    allocation.distributedRebates = BigInt.fromI32(0)
+    allocation.curatorRewards = BigInt.fromI32(0)
+    allocation.indexingRewards = BigInt.fromI32(0)
+    allocation.indexingIndexerRewards = BigInt.fromI32(0)
+    allocation.indexingDelegatorRewards = BigInt.fromI32(0)
+    allocation.delegationFees = BigInt.fromI32(0)
+    allocation.status = 'Active'
+    allocation.totalReturn = BigDecimal.fromString('0')
+    allocation.annualizedReturn = BigDecimal.fromString('0')
+    allocation.createdAt = event.block.timestamp.toI32()
+    allocation.indexingRewardCutAtStart = provision.indexingRewardsCut.toI32()
+    allocation.indexingRewardEffectiveCutAtStart = provision.indexingRewardsCut.toBigDecimal()
+    allocation.queryFeeCutAtStart = provision.queryFeeCut.toI32()
+    allocation.queryFeeEffectiveCutAtStart = provision.queryFeeCut.toBigDecimal()
```

Comment:
same, i think it's incorrect and we need to calculate

Reply:
Actually no need to calculate, but rather use the actual latest calculated value from the provision 😅 .

Also thanks to this I realized the fields we were using for the AllocationClosed case were the same as the ones we were using for the AllocationCreated case (i.e. we were using `indexingRewardEffectiveCutAtStart` on both, instead of `indexingRewardEffectiveCutAtStart` for Created, and `indexingRewardEffectiveCutAtClose` for Closed 😅)